### ISSUE TITLE
Fury should be able to build itself

### ISSUE THREAD

**Issue by rberenguel** [propensive/fury-old#56]
And probably messing up horribly. From inside fury's git clone

```
fury init
fury project add -n fury
fury repo add -r gh:propensive/base
fury import add -i base:2.12.6
fury module update -c scala/compiler
fury source add -d local:src/build
fury source add -d local:src/core
fury source add -d local:src/dependency
fury source add -d local:src/imports 
fury source add -d local:src/main/scala-2.12
fury source add -d local:src/menu
fury source add -d local:src/module # and I stop here because I start getting duplicated classes (?)

fury repo add -r gh:propensive/guillotine

fury
```

```
not found: object guillotine
[E]       L18: import guillotine._
```

But now running fury gets into bloop and the compiler does not seem to know where to find guillotine. I missed/messed something, for sure :)

**Comment by rberenguel** [comment]
Changed to creating modules for each `src/folder` and adding each to its own as a source as suggested by @propensive . Switched to `core`, but still could not find guillotine. 

More worryingly, I added strings as a dependency of `core` (since it needs MsgShow, so I thought it was a good addition to see how it incrementally improved) and looks like `strings` was totally fine, _but_ `core` could not find what it defined (`MsgShow` is defined in `strings`)

```
 » fury/strings    .  [*]
 └─» fury/core        [!]
   └─» fury/main   .  [ ]
[...]
[E] [E49] ../src/core/package.scala:45:32
[E]       not found: type MsgShow
[E]       L45:   implicit val msgShowBoolean: MsgShow[Boolean] = if(_) msg">" else msg""
```
(note this is not the first appearance of this type as missing). To add weirdness, there's a `MsgShow.class` present in `.fury`

Note: I tried running `fury repo pull --repo (any)` but all end with `fatal: this operation must be run in a work tree`.
